### DISCUSSION TITLE
Schema naming in openapi.json

### DISCUSSION THREAD

**Discussion by pboguslawski** [swaggest/rest#123] [Q&A] [answered]
It is possible to force operation names with `u.SetName("myApiOp")` which sets `operationId` in `openapi.json`. Struct names in `openapi.json` follow `PackagenameStructname` convention from go source.

Is there any way to force struct name in `openapi.json`? If not - maybe removing `Packagename` part is possible (structure name in `openapi.json` identical with struct name in go source)?

**Comment by vearutop** [comment]
Not sure if I properly understand the question, but if you want to apply changes to OpanAPI operation, that is created from usecase, you can use annotation on `*nethttp.Handler` (like [here](https://github.com/swaggest/rest/blob/v0.2.37/_examples/advanced-generic/router.go#L135-L141)):

```go        
	s.Get("/foo", fooUsecase(), func(h *nethttp.Handler) {
		h.OperationAnnotations = append(h.OperationAnnotations, func(op *openapi3.Operation) error {
			op.WithID("foo")

			return nil
		})
	})
```

**Comment by pboguslawski** [comment]
>  but if you want to apply changes to OpanAPI operation

My question is about go structs (openapi schemas) not go functions (openapi operations).

For example go struct

```
package items

type Item struct {
_ struct{} `title:"Item"`
 Name string
}
```
becomes `ItemsItem` in `openapi.json` schemas; is it possible to force this schema name in `openapi.json` to be `Item` not `ItemsItem`?

Question is about changing schema name not title in `openapi.json` i.e.

```
[...]
 "components": {
  "schemas": {
   "ItemsItem": {  <<<<<<<< how to force this name in go definition i.e. Item not ItemsItem?
    "title": "Item",  <<< this one is only title that may be changed using title tag in go definition
[...]
```

**Comment by vearutop** [comment, marked as answer]
This customization can be done with `InterceptDefName` hook.
```go
	s.OpenAPICollector.Reflector().InterceptDefName(func(t reflect.Type, defaultDefName string) string {
		return strings.ReplaceAll(defaultDefName, "Items", "")
	})
```

**Reply by pboguslawski** [reply]
Works fine thank you! Consider allowing name changing with go code/tag as alternative to hook.
